syntax = "proto3";

package tokenservice;

option go_package = "tokenservice";

service TokenService {
  rpc GetToken (TokenRequest) returns (TokenResponse) {}
}


message TokenRequest {
  string requestId = 1;
  string processID = 2;
}

message Secret {
  string name = 1;
  SecretType type = 2;
  bytes data = 3;

  enum SecretType {
    RAW = 0;
    TPM = 1;
    TINK = 2;
  }  
}

message TokenResponse {
  string responseID = 1;
  string inResponseTo = 2;
  repeated Secret secrets = 3;
}
